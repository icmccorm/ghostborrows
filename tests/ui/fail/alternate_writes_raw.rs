use ghostborrows::*;

fn main() {
    let value = Value::new(0);
    value.borrow_mut(|x, token| {
        let pair = x.split();
        let y = RefReserved::from(pair);
        let z = RefReserved::from(pair);
        *y.activate(token) += 1;
        *z.activate(token) += 1;
    });
}
