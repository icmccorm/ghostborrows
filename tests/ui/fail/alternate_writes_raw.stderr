error[E0382]: use of moved value: `token`
  --> tests/ui/fail/alternate_writes_raw.rs:10:21
   |
5  |     value.borrow_mut(|x, token| {
   |                          ----- move occurs because `token` has type `Token<'_, i32>`, which does not implement the `Copy` trait
...
9  |         *y.activate(token) += 1;
   |                     ----- value moved here
10 |         *z.activate(token) += 1;
   |                     ^^^^^ value used here after move
